packages/autorest.python/autorest/codegen/models/base.py

```diff
@@ -7,6 +7,25 @@
 from abc import ABC, abstractmethod
 from .imports import FileImport
 
+# pylint: disable=line-too-long
+SIGNATURE_CONVERT = {
```

Comment:
If there is no if_none_match, then this will not work. I don't think you have implemented a common signature convert logic since this only work for parameter 1 to 1 replacement. I prefer to add this logic to param list parser. When we meet any conditional request header, we remove them all and add two pre-defined parameters.

Reply:
There are 2 reasons why I choose current design:
(1) I find no scenario that there are only one of if-match and if-none-match
(2) Only operation signature will be influenced and builder signature keeps same as defined. So `signature convert` is the easiest implementation.

In view of that only etag needs this kind of convert, I don't pretend to design a common pattern for it. If there is more scenarios that need it, it is not late to refactor it. 

add @iscai-msft for awareness.

Reply:
My point is it is better to change the signature in model level instead of template level. IOW, the transform could be limited in parameter list level, instead of adding convert config in base model level and do the change in serialization level.

Reply:
No matter which pattern to choose, it is necessary to do some change in serialization level. For example, there are several place that need client_name and its default value, however, they have different value when serialization in different place (operation/builder/call_builder). At least for now, it is unnecessary to do big refactor for this feature.

Reply:
I have a discussion with Izzy this morning about varies kinds of special logic of headers. We come to the agreement that we should better do it in pre-processor and make the model and generation logic clean. What do you think? cc: @iscai-msft